src/System.Drawing.Common/src/System/Drawing/Font.cs

```diff
@@ -275,8 +275,70 @@ public override string ToString()
                                     _gdiCharSet,
                                     _gdiVerticalFont);
         }
-        
+
         // This is used by SystemFonts when constructing a system Font objects.
         internal void SetSystemFontName(string systemFontName) => _systemFontName = systemFontName;
+
+        public unsafe void ToLogFont(object logFont, Graphics graphics)
+        {
+            if (logFont == null)
+            {
+                throw new ArgumentNullException(nameof(logFont));
+            }
+
+            Type type = logFont.GetType();
+            int nativeSize = sizeof(SafeNativeMethods.LOGFONT);
+            if (Marshal.SizeOf(type) != nativeSize)
+            {
+                // If we don't actually have an object that is LOGFONT in size, trying to pass
+                // it to GDI+ is likely to cause an AV.
+                throw new ArgumentException();
+            }
+
+            SafeNativeMethods.LOGFONT nativeLogFont = ToLogFontInternal(graphics);
+
+            if (!type.IsValueType)
+            {
+                // Unfortunately PtrToStructure requires that the passed in object not be a value type.
+                // The underlying runtime code supports it, and perhaps there is another way to do this.
```

Comment:
Blasting a new content over a boxed valuetype is unsafe operation. That's why the `PtrToStructure` APIs are designed they way they are designed - to make them somewhat more safe. I think this comment can be deleted. It is pretty unlikely we would ever try to fix that in these legacy APIs.

Reply:
Good to know, I'll update it.